### PR TITLE
Improve support of assertArrayHasKey()

### PR DESCRIPTION
**Pull request by villfa**
Related to #167

### CODE REVIEW

File: tests/Type/PHPUnit/data/assert-function.php [hunk 1]
```diff
@@ -20,15 +20,21 @@ public function doFoo($o): void
 		assertType(self::class, $o);
 	}
 
-	public function arrayHasNumericKey(array $a): void {
+	public function arrayHasNumericKey(array $a, \ArrayAccess $b): void {
 		assertArrayHasKey(0, $a);
 		assertType('array&hasOffset(0)', $a);
+
+		assertArrayHasKey(0, $b);
+		assertType('ArrayAccess', $b);
```
Comment (villfa) [inline tests/Type/PHPUnit/data/assert-function.php:39]: I wanted to obtain this result but for some reasons it didn't work. However before my changes the result was `*NEVER*` so it's already an improvement.
```suggestion
		assertType('ArrayAccess&hasOffset(0)', $b);
```

**Comment by ondrejmirtes** [comment]
Sorry for the late review - please rebase because of a conflict. Thank you.

As for your suggestion - HasOffsetType and HasOffsetValueType support only intersection with arrays.

**Comment by villfa** [comment]
The branch is now up to date.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**